stages/org.osbuild.coreos.live-artifacts.mono

```diff
@@ -510,133 +642,19 @@ def main(workdir, tree, inputs, options, loop_client):
-        os.truncate(iso_initramfs, initramfs_size + IGNITION_IMG_SIZE)
-        subprocess.check_call(['/usr/bin/mk-s390image',
-                               kernel_dest,
-                               os.path.join(tmpisoimages, 'cdboot.img'),
-                               '-r', iso_initramfs,
-                               '-p', os.path.join(tmpisoimages, 'cdboot.prm')])
-        os.truncate(iso_initramfs, initramfs_size)
-
-        # Get the kargs and initramfs offsets in the cdboot.img. For more info, see:
-        # https://github.com/ibm-s390-linux/s390-tools/blob/032304d5034e/netboot/mk-s390image#L21-L23
-        CDBOOT_IMG_OFFS_INITRD_START_BYTES = 66568
-        CDBOOT_IMG_OFFS_KARGS_START_BYTES = 66688
-        CDBOOT_IMG_OFFS_KARGS_MAX_SIZE = 896
-        with open(os.path.join(tmpisoimages, 'cdboot.img'), 'rb') as f:
-            f.seek(CDBOOT_IMG_OFFS_INITRD_START_BYTES)
-            offset = struct.unpack(">Q", f.read(8))[0]
-
-            # sanity-check we're at the right spot by comparing a few bytes
-            f.seek(offset)
-            with open(iso_initramfs, 'rb') as canonical:
-                if f.read(1024) != canonical.read(1024):
-                    raise ValueError(f"expected initrd at offset {offset}")
-
-            igninfo_json = {
-                'file': 'images/cdboot.img',
-                'offset': offset + initramfs_size,
-                'length': IGNITION_IMG_SIZE,
-            }
-
-            # kargs are part of 'images/cdboot.img' blob
-            kargs_json['files'].append({
-                'path': 'images/cdboot.img',
-                'offset': CDBOOT_IMG_OFFS_KARGS_START_BYTES,
-                'pad': '\0',
-                'end': '\0',
-            })
-            kargs_json.update(
-                size=CDBOOT_IMG_OFFS_KARGS_MAX_SIZE,
-            )
-        # generate .addrsize file for LPAR
-        with open(os.path.join(tmpisoimages, 'initrd.addrsize'), 'wb') as addrsize:
-            addrsize_data = struct.pack(">iiii", 0, int(INITRD_ADDRESS, 16), 0,
-                                        os.stat(iso_initramfs).st_size)
-            addrsize.write(addrsize_data)
-
-        # safely remove things we don't need in the final ISO tree
-        for d in ['EFI', 'isolinux']:
-            shutil.rmtree(os.path.join(tmpisoroot, d))
-
-        genisoargs = ['/usr/bin/xorrisofs', '-verbose',
-                      '-volid', volid,
-                      '-volset', f"{name_version}",
-                      '-rational-rock', '-J', '-joliet-long',
-                      '-no-emul-boot', '-eltorito-boot',
-                      os.path.join(os.path.relpath(tmpisoimages, tmpisoroot), 'cdboot.img')]
+        genisoargs = genisoargs_s390x(tmpisoimages, tmpisoroot, tmpisoimagespxe, kernel_img, test_fixture,
+                                      iso_initramfs, volid, name_version)
+        offset = get_cdbootimg_offset(tmpisoimages, iso_initramfs, kargs_json)
```

Comment:
maybe we just move the `igninfo_json` update below into this function as well and rename it slightly?

```
update_image_offsets(tmpisoimages, iso_initramfs, kargs_json, igninfo_json)
```